httpclient5/src/test/java/org/apache/hc/client5/http/examples/ClientServerCompressionExample.java

```diff
@@ -0,0 +1,186 @@
+ * {@link org.apache.hc.client5.http.entity.EntityBuilder#compressed
+ * EntityBuilder#compressed(ContentCoding)} API can be used to
+ * transparently <strong>compress</strong> a request entity with
+ * Zstandard (<em>zstd</em>) on the client side and then
+ * <strong>decompress</strong> it on the server side—while the reverse
+ * happens for the response.</p>
+ *
+ * <h3>What the code does</h3>
+ * <ol>
+ *   <li>Spins up a tiny in-JVM {@code com.sun.net.httpserver.HttpServer}
+ *       bound to an ephemeral port.</li>
+ *   <li>The client sends a {@code POST /echo} whose body is compressed
+ *       with <em>zstd</em> simply by calling
+ *       {@code compressed(ContentCoding.ZSTD)}.</li>
+ *   <li>The server handler decodes the request body with Commons Compress
+ *       ({@link org.apache.commons.compress.compressors.CompressorStreamFactory}),
+ *       echoes the text back, re-encodes it with Zstandard, and sets
+ *       {@code Content-Encoding: zstd}.</li>
+ *   <li>The client receives the response—HttpClient 5.6+ notices the
+ *       header, automatically picks the decoder you registered via
+ *       {@link org.apache.hc.client5.http.entity.compress.ContentDecoderRegistry},
+ *       and hands the caller the already-decompressed text.</li>
+ * </ol>
+ *
+ * <h3>How to run</h3>
+ * <ul>
+ *   <li>Java 8-17 (only standard {@code com.sun.net.httpserver} API used).</li>
+ *   <li>Maven dependencies (all <em>compile + runtime</em>):<br>
+ *     ─ {@code httpclient5 ≥ 5.6-SNAPSHOT}<br>
+ *     ─ {@code commons-compress ≥ 1.21}<br>
+ *     ─ {@code com.github.luben:zstd-jni} (automatically pulled by
+ *       Commons Compress for the native codec)
+ *   </li>
+ *   <li>JDK 17+ users: you will see a one-time warning that
+ *       {@code zstd-jni} loads native code. Add<br>
+ *       {@code --enable-native-access=ALL-UNNAMED} to the VM options
+ *       if you want to silence it.</li>
+ *   <li>Launch from your IDE or:
+ *   <pre>{@code
+ *   mvn -q exec:java \
+ *       -Dexec.mainClass=org.apache.hc.client5.testing.classic.ZstdRoundTrip
+ *   }</pre></li>
+ * </ul>
+ *
+ * <h3>Why Zstandard and not GZIP?</h3>
+ * GZIP support has always been built-in.  This demo illustrates the
+ * new <em>pluggable</em> compression framework: as soon as
+ * Commons Compress + the native helper JAR are on the class-path,
+ * HttpClient discovers the codec and you can opt-in at the
+ * <cite>EntityBuilder</cite> call-site—no other code changes needed.
+ *
+ * @since 5.6
+ */
+public final class ClientServerCompressionExample {
+
+    private ClientServerCompressionExample() {
+    }
+
+    public static void main(final String[] args) throws Exception {
+
+        /* ──────────────────────────────────────────────
+           1. Tiny echo server that understands “br|zstd…”
+           ────────────────────────────────────────────── */
+        final HttpServer server = HttpServer.create(new InetSocketAddress(0), 0);
```

Comment:
@arturobernalg Would not it be nicer to use our own `HttpServer` from core here?